Realm/ObjectStore/shared_realm.cpp

```diff
@@ -127,33 +128,8 @@ void Realm::init(std::shared_ptr<RealmCoordinator> coordinator)
 
     try {
         // otherwise get the schema from the group
-        auto target_schema = std::move(m_config.schema);
-        auto target_schema_version = m_config.schema_version;
         m_config.schema_version = ObjectStore::get_schema_version(read_group());
         m_config.schema = std::make_unique<Schema>(ObjectStore::schema_from_group(read_group()));
-
-        // if a target schema is supplied, verify that it matches or migrate to
-        // it, as neeeded
-        if (target_schema) {
-            if (m_config.read_only) {
```

Comment:
This logic does all need to exist, and it's somewhat concerning that apparently there are no tests for it.